deepinterpolation/generator_collection.py

```diff
@@ -897,112 +900,144 @@ class OphysGenerator(SequentialGenerator):
 
-        input_index = np.arange(
-            index_frame - self.pre_frame - self.pre_post_omission,
-            index_frame + self.post_frame + self.pre_post_omission + 1,
-        )
-        input_index = input_index[input_index != index_frame]
-
-        for index_padding in np.arange(self.pre_post_omission + 1):
-            input_index = input_index[input_index !=
-                                      index_frame - index_padding]
-            input_index = input_index[input_index !=
-                                      index_frame + index_padding]
-
-        # If data was cached we do not need to normalize. this was done 
-        # at once to minimize compute
-        if self.cache_data:
-            data_img_input = movie_obj[input_index, :, :]
-            data_img_output = movie_obj[index_frame, :, :]
+    def __getitem__(self, index):
+        if self.gpu_available:
+            if self.gpu_cache_full:
+                batch_indices = self.generate_batch_indexes(index)
+                input_indices = np.vstack(
+                    [self.__index_generation__(frame_index) \
+                        for frame_index in batch_indices])
+            else:
+                self.__gpu_cache__(index)
+                input_indices = self.input_indices
+                batch_indices = self.batch_indices
+
+            input_full = tf.gather(self.data_tensor, input_indices)
+            output_full = tf.gather(self.data_tensor, batch_indices)
+            input_full = tf.experimental.numpy.moveaxis(input_full, 1, -1)
+            output_full = tf.expand_dims(output_full, -1)
         else:
-            data_img_input = (movie_obj[input_index, :, :].astype("float") \
-                - self.local_mean ) / self.local_std
-            data_img_output = (movie_obj[index_frame, :, :].astype("float") \
-                - self.local_mean ) / self.local_std
+            batch_indices = self.generate_batch_indexes(index)
+            input_indices = np.vstack(
+                [self.__index_generation__(frame_index) \
+                    for frame_index in batch_indices])
+            
+            input_full = self.movie_data[input_indices].astype("float")
+            output_full = self.movie_data[batch_indices].astype("float")
+            if not self.normalize_cache:
+                input_full = normalize(input_full, self.local_mean,
+                    self.local_std)
+                output_full = normalize(output_full, self.local_mean,
+                    self.local_std)
+            input_full = np.moveaxis(input_full, 1, -1)
+            output_full = np.expand_dims(output_full, -1)
 
-        data_img_input = np.swapaxes(data_img_input, 1, 2)
-        data_img_input = np.swapaxes(data_img_input, 0, 2)
-
-        if not self.cache_data:
-            movie_obj_point.close()
+        return input_full, output_full
 
-        return data_img_input, data_img_output
+    def __index_generation__(self, index_frame):
```

Comment:
Only "magic" methods should use `__ __`. please make private instead. Also method names should start with an action verb, such as `get`. Also the name for this method can be made more clear as to "what" indices are being generated such as `_get_indices_for_frame_block`